src/main.py

```diff
@@ -0,0 +1,119 @@
+# (c) Open Compute Project
+# (c) Google LLC
+# (c) Igor Semenov (goshik92@gmail.com)
+# 
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+import ocptv.output as tv
+import argparse
+import sh
+import re
+import os
+import contextlib
+from memtester_parsing import MemtesterObserver
+
+def main():
+    parser = argparse.ArgumentParser(description="OCP-compliant memtester wrapper.")
+    parser.add_argument("--mt_path", default="memtester",
+        help="Path to memtester executable")
+    parser.add_argument("--mt_args", default="",
+        help="Memtester arguments packed into a string. See `man memtester` for more detail")
+    parser.add_argument("--mt_log_filename", default=None,
+        help="Path to the file that will contain raw memtester output (both STDOUT and STDERR)")
+    args = parser.parse_args()
+
+    run = tv.TestRun(name="memtester", version="1.0")
+    dut = tv.Dut(id=sh.hostid().strip(), name=sh.hostname().strip())
+    observer = MemtesterObserver()
+    
+    # Use ExitStack() to avoid deep nesting
+    with contextlib.ExitStack() as estack:
+        estack.enter_context(run.scope(dut=dut))
+        estack.enter_context((step := run.add_step("run-memtester")).scope())
+
+        bad_addrs = step.start_measurement_series(name="bad_addrs")
+        
+        # Check if a supported version of memtester is installed in the system
+        def version_callback(version, is_known):
```

Comment:
the determination of whether this version is acceptable should be done before the `tv.run` begins. This does indeed result in running the binary twice (first with `--version`) but i find that acceptable and this
1. allows the user to say "i refuse to run this if not a supported version"
2. the version of memtester to be correctly reported in dut.SoftwareInfos, rather than a log artifact

Reply:
https://github.com/opencomputeproject/ocp-diag-memtester/issues/4